### PR TITLE
Dev/issue 8631 rest v3

### PR DESCRIPTION
**Pull request by mcantelon**
New REST endpoints (some copied from our Binder project), and modification of current endpoints, to provide similar functionality to ArchivesSpace's REST endpoints (in support of our agentarchives Python client).

### CODE REVIEW

File: plugins/arRestApiPlugin/modules/api/actions/digitalobjectsBrowseAction.class.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        $criteria->addJoin(QubitTerm::ID, $do->mediaTypeId);
+
+        if (null !== ($typeTerm = QubitTerm::getOne($criteria)))
+        {
+          $fields['media_type'] = $typeTerm->name;
+        }
+      }
+
+      // Add related property data
+      $propertyFields = array(
```
Comment (MikeFE) [inline plugins/arRestApiPlugin/modules/api/actions/digitalobjectsBrowseAction.class.php:69]: probably no need to do a bunch of line breaks for code <120 characters?

File: plugins/arRestApiPlugin/modules/api/actions/digitalobjectsBrowseAction.class.php [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        if (null !== ($uuidProperty = QubitProperty::getOne($criteria)))
+        {
+          $fields[$apiFieldLabel] = $uuidProperty->value;
+        }
+      }
+
+      $results[$do->id] = $fields;
+    }
+
+    return
```
Comment (MikeFE) [inline plugins/arRestApiPlugin/modules/api/actions/digitalobjectsBrowseAction.class.php:86]: same here

File: plugins/arRestApiPlugin/modules/api/actions/digitalobjectsCreateAction.class.php [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        $criteria = new Criteria;
+        $criteria->addJoin(QubitTerm::ID, QubitTermI18n::ID);
+        $criteria->add(QubitTerm::TAXONOMY_ID, QubitTaxonomy::MEDIA_TYPE_ID);
+        $criteria->add(QubitTermI18n::NAME, $value);
+        $typeTerm = QubitTerm::getOne($criteria);
+        $this->do->mediaType = $typeTerm;
+        break;
+
+      case 'url':
+
```
Comment (MikeFE) [inline plugins/arRestApiPlugin/modules/api/actions/digitalobjectsCreateAction.class.php]: probably no need for this case if all it does is break out

**Comment by sevein** [comment]
8631 is a private ticket! Please fix!

**Comment by mcantelon** [comment]
Merged into qa/2.4.x.

**Pull request closed without merging**